### PR TITLE
Fix various problems obscured in default R Hector

### LINKED ISSUE #365: Check X set up for C++
Following up from our conversation on May 13 you asked if I could create an issue for you to check the X code from the C++ hector. 

Also idk if you have time but may be we should put together some instructions for how to run hector using C++ without the R interface unless you don't think that is useful 🤷‍♀️

### LINKED ISSUE #372: Caller is requesting unknown variable: hc_concentration
Something goofy has developed in the Hector command-line build; maybe it's a code problem, maybe it's an instructions problem, but either way it's a problem. I'm concerned that with all the focus on the R package over the last 1-2 years we've neglected the core (command line) configuration.

> Well, Dominique, I owe you an apology. This morning I built the Hector master, updating the Xcode project settings (see PR Kalyn), and initially got the same error—
>  
> Caller is requesting unknown variable: hc_concentration
>  
> I didn’t understand this; the R and command-line versions are built from the same sources, and if the former runs, the latter should as well.
>  
> BUT after quitting and then restarting Xcode, and rebuilding everything…Hector ran without a problem:
>  
> Thu May 21 05:33:17 2020:NOTICE:run_spinup: Carbon model is spun up after 352 steps
> Thu May 21 05:33:17 2020:NOTICE:run_spinup: Model spun up after 352 steps
> Thu May 21 05:33:17 2020:NOTICE:main: Running the core.
> Thu May 21 05:33:17 2020:NOTICE:run: Running...
> Thu May 21 05:33:19 2020:NOTICE:main: Hector wrapper end
> Program ended with exit code: 0
>  
> Good, but now I really don’t understand! We definitely have a problem, even though I’m not yet sure what it is.
>  
> Compounding things, I took a careful look at the Travis logs, and test_hector.sh is broken but isn’t generating the error code that would have caused Travis to fail and thus alert us. [ this is #371 ]

We need to figure out what's going on.

### LINKED ISSUE #381: Package code tests need to test with suppresslogging = FALSE
See #372

### PR DESCRIPTION
**Pull request by bpbond**
...that crept in because of broken Travis test script (see #373 ). Specifically:

* Code to handle "CF4_halocarbon" in `halocarbon_component.cpp` restored
* Corrected handling of D_LAND_CFLUX and D_OCEAN_CFLUX in `simpleNbox.cpp` and `ocean_component.cpp` respectively
* Created new atmosphere-land flux variable in `simpleNbox.hpp`
* Updated Xcode project file to use environment variables, not absolute paths, for Boost
* Fixed pathnames to executable in `test_hector.sh` for both gcc and Clang
* Added an R package test with logging turned ON

Closes #365 
Closes #372 
Closes #381

### CODE REVIEW

**Comment by bpbond** [comment]
@kdorheim Do you have time to take a hard look at model outputs (via R package is fine) and confirm things look OK? 🙏

File: tests/testthat/test_hector.R [hunk 1]
```diff
@@ -22,6 +22,11 @@ test_that('Basic hcore functionality works', {
     hc <- shutdown(hc)
     expect_false(isactive(hc))
 
+    ## Turn logging ON for one test and confirm it runs (see #372 and #381)
+    hc_log <- newcore(file.path(inputdir, 'hector_rcp45.ini'), name='RCP45', suppresslogging = FALSE)
+    run(hc_log, 2100)
+    shutdown(hc_log)
+
```
[the remark this answers is not in the document]
Comment (bpbond) [reply tests/testthat/test_hector.R:29]: Good idea, thanks.

**Comment by bpbond** [comment]
Re the vignette rebuild erroring, it seems to do fine on Travis...where are you seeing this? On your local machine? If so you may need to install the `nleqslv` package? Anyway, if you're seeing the error online please point me to it--thanks.

**Comment by kdorheim** [comment]
Installing nleqslv did the trick.

**Comment by bpbond** [comment]
Great. Well, let me know what else needs doing here please, or approve and merge if you think ready. Thanks.

**Pull request merged**